File: php/fibonacci/fibonacci.php
<?php

function fibonacci($n) {
    if ($n == 0 || $n == 1) {
        return $n;
    } else {
        return fibonacci($n-1) + fibonacci($n-2);
    }
}

function fibonacciLine($n) {
    $list = array();
    for ($x = 0; $x < $n; $x++) {
        $list[] = fibonacci($x);
    }
    return $list;
}

foreach (fibonacciLine(10) as $value) {
    echo $value . "\n";
}

File: scala/fibonacci/Fibonacci.scala

object Fibonacci {
    def main(args: Array[String]) {
        for (x <- fibonacciLine(10)) {
            println(x)
        }
    }

    def fibonacciLine(n: Int): List[Int] = {
        return (0 to n-1).map(x => fibonacci(x)).toList
    }

    def fibonacci(n: Int): Int = {
        if (n == 0 || n == 1) return n
        else return fibonacci(n-1) + fibonacci(n-2)
    }
}

